lib/ecto/adapters/mssql/connection.ex

```diff
@@ -0,0 +1,1644 @@
+        else
+          [
+            ?\s,
+            ?(,
+            intersperse_map(header, ", ", &quote_name/1),
+            ?),
+            returning,
+            " VALUES " | insert_all(rows, 1)
+          ]
+        end
+
+      ["INSERT INTO ", quote_table(prefix, table), values]
+    end
+
+    defp on_conflict({_, _, [_ | _]}, _header) do
+      error!(nil, "The :conflict_target option is not supported in insert/insert_all by MsSql")
+    end
+
+    defp on_conflict({:raise, _, []}, _header) do
+      []
+    end
+
+    defp on_conflict({:nothing, _, []}, [_field | _]) do
+      error!(nil, "The :nothing option is not supported in insert/insert_all by MsSql")
+    end
+
+    defp on_conflict({:replace_all, _, []}, _header) do
+      error!(nil, "The :replace_all option is not supported in insert/insert_all by MsSql")
+    end
+
+    defp on_conflict({_query, _, []}, _header) do
+      error!(
+        nil,
+        "The query as option for on_conflict is not supported in insert/insert_all by MsSql yet."
+      )
+    end
+
+    defp insert_all(rows, counter) do
+      intersperse_reduce(rows, ",", counter, fn row, counter ->
+        {row, counter} = insert_each(row, counter)
+        {[?(, row, ?)], counter}
+      end)
+      |> elem(0)
+    end
+
+    defp insert_each(values, counter) do
+      intersperse_reduce(values, ", ", counter, fn
+        nil, counter ->
+          {"DEFAULT", counter}
+
+        {%Query{} = query, params_counter}, counter ->
+          {[?(, all(query), ?)], counter + params_counter}
+
+        _, counter ->
+          {[?@ | Integer.to_string(counter)], counter + 1}
+      end)
+    end
+
+    @impl true
+    def update(prefix, table, fields, filters, returning) do
+      {fields, count} =
+        intersperse_reduce(fields, ", ", 1, fn field, acc ->
+          {[quote_name(field), " = @", Integer.to_string(acc)], acc + 1}
+        end)
+
+      {filters, _count} =
+        intersperse_reduce(filters, " AND ", count, fn
+          {field, nil}, acc ->
+            {[quote_name(field), " IS NULL"], acc + 1}
+
+          {field, _value}, acc ->
+            {[quote_name(field), " = @", Integer.to_string(acc)], acc + 1}
+
+          field, acc ->
+            {[quote_name(field), " = @", Integer.to_string(acc)], acc + 1}
+        end)
+
+      [
+        "UPDATE ",
+        quote_table(prefix, table),
+        " SET ",
+        fields,
+        returning(returning, "INSERTED"),
+        " WHERE " | filters
+      ]
+    end
+
+    @impl true
+    def delete(prefix, table, filters, returning) do
+      {filters, _} =
+        intersperse_reduce(filters, " AND ", 1, fn
+          {field, nil}, acc ->
+            {[quote_name(field), " IS NULL"], acc + 1}
+
+          {field, _value}, acc ->
+            {[quote_name(field), " = @", Integer.to_string(acc)], acc + 1}
+
+          field, acc ->
+            {[quote_name(field), " = @", Integer.to_string(acc)], acc + 1}
```

Comment:
You can remove this clause, both clauses above are enough.

Reply:
Actually I need this for [this test](https://github.com/mjaric/ecto_sql/blob/8ed7fa0bfef62572e7a485451575c4f02a8ee2e2/test/ecto/adapters/mssql_test.exs#L790)

Reply:
Those tests are out of date. :) See the equivalent tests for PG: https://github.com/mjaric/ecto_sql/blob/8ed7fa0bfef62572e7a485451575c4f02a8ee2e2/test/ecto/adapters/postgres_test.exs#L1103-L1115